### DISCUSSION TITLE
Relationship declaration in 2.0

### DISCUSSION THREAD

**Discussion by liverpool1026** [sqlalchemy/sqlalchemy#10830] [Usage Questions] [answered]
Hi there,

We are in the process of migrating from 1.4 -> 2.0.

I think that came to my mind is that relationship seems to be returning 1 item when it is even a 1 to many relationship.

In 1.4, this works fine
```
Class Test:
  ....
  bounds  = sa.orm.RelationshipProperty = relationship("Bounds", viewonly=True)

  
for bound in test.bounds:
   ....

```

But in 2.0, this only returns one obj when accessing bounds instead of list.

Looking at the docs, it did mention that we can type it using Mapped + Python type collection
![image](https://github.com/sqlalchemy/sqlalchemy/assets/29122286/973984a2-f172-4d37-9972-4d387db151a6)

But that is when there is no argument passed into relationship(). Wondering to handle one to many relationship, do we always need to add the list typing in the front?

bounds  = sa.orm.Mapped[list["Bounds"]] = relationship("Bounds", viewonly=True) -> This works, but I am passing arguments as well, just checking if the "list" is required for 1-to-many or there are another way of defining it.

Thanks

**Comment by CaselIT** [comment]
Hi,

This does not look correct:
`bounds  = sa.orm.RelationshipProperty = relationship("Bounds", viewonly=True)` 
Did you mean
`bounds: sa.orm.RelationshipProperty = relationship("Bounds", viewonly=True)`?

You can manually set `uselist` in any case

**Reply by liverpool1026** [reply]
yep, made a typo there.

Cool, uselist is what I am looking for, but reading the docs 
![image](https://github.com/sqlalchemy/sqlalchemy/assets/29122286/2a987c44-5dea-4281-89c9-100c9aa1eff8)

It says it can be automatically determined for most time. But it looks like it is not automatically detecting 1-to-many relationship in 2.0 for my code. Anything special changed here?

**Reply by zzzeek** [reply, marked as answer]
it's not correct even in 1.4 to annotate the mapping like this:

```
class Test(Base):
     # ...
     x: RelationshipProperty = relationship(...)

```

that's not what's actually on your `Test` class as soon as the class is constructed; class instrumentation replaces that `relationship()` with `InstrumentedAttribute` which is a Python descriptor.

this is why the annotation, if present should use `Mapped`:

```
class Test(Base):
     # ...
     x: Mapped[List[Bound]] = relationship(...)

```


otherwise yes, the ORM will see `RelationshipProperty` and not see that there's any collection set up.  it really should not be accepting that annotation at all, im not really sure why it is.

**Reply by liverpool1026** [reply]
Thanks both of you for the response.
